### ISSUE TITLE
Docker stuck at Starting Engine, becomes unresponsive

### ISSUE THREAD

**Issue by AachoLoya** [docker/for-win#13526]
### Description

Been experiencing this issue for a while on two different installations of Windows Server 2022. I upgraded Docker to 4.19 and also updated wsl Ubuntu to latest and updated packages as well. Since then, I'm getting issues with Docker stuck on "Starting the Docker Engine..." and eventually it fails with following output:

`System.Exception:
1 error occurred:
	* waiting for docker daemon: context deadline exceeded

   at Docker.Engines.WSL2.LinuxWSL2Engine.<DoStartAsync>d__10.MoveNext() in C:\workspaces\PR-22253\src\github.com\docker\pinata\win\src\Docker.Engines\WSL2\LinuxWSL2Engine.cs:line 93
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.TaskExtensions.<WrapAsyncInCancellationException>d__0.MoveNext() in C:\workspaces\PR-22253\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\TaskExtensions.cs:line 29
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__7.MoveNext() in C:\workspaces\PR-22253\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 79
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__7.MoveNext() in C:\workspaces\PR-22253\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 108
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Docker.Engines.Engines.<WaitCurrentTransitionsAsync>d__30.MoveNext() in C:\workspaces\PR-22253\src\github.com\docker\pinata\win\src\Docker.Engines\Engines.cs:line 290
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Engines.Engines.<StartAsync>d__24.MoveNext() in C:\workspaces\PR-22253\src\github.com\docker\pinata\win\src\Docker.Engines\Engines.cs:line 114
`

Following is the output from wsl --version command:

`WSL version: 1.1.3.0
Kernel version: 5.15.90.1
WSLg version: 1.0.49
MSRDC version: 1.2.3770
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.20348.1726`

### Reproduce

Updated Docker Desktop to 4.19 and 4.20 (having issues on both versions). Updated ubuntu/wsl to latest version.

### docker version

```bash
error during connect: in the default daemon configuration on Windows, the docker client must be run with elevated privileges to connect: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.24/version": open //./pipe/docker_engine: The system cannot find the file specified.
Client:
 Cloud integration: v1.0.33
 Version:           24.0.2
 API version:       1.43
 Go version:        go1.20.4
 Git commit:        cb74dfc
 Built:             Thu May 25 21:53:15 2023
 OS/Arch:           windows/amd64
 Context:           default
```

### docker info

```bash
ERROR: error during connect: in the default daemon configuration on Windows, the docker client must be run with elevated privileges to connect: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.24/info": open //./pipe/docker_engine: The system cannot find the file specified.
errors pretty printing info
```

### Diagnostics ID

B88E2B36-325A-4E27-ABB2-D4AB3AC81799/20230602181618

**Comment by Darkbelg** [comment]
Can you check the disk size of your virtual hdd?

**Comment by eric-gitta-moore** [comment]
- docker desktop 4.22.0
- docker engine 24.0.5
- Win11 22H2 clean install

This problem also arises, "starting“

Any news?

**Comment by AachoLoya** [comment]
After spending two days trying to figure out this issue, I just removed it altogether and setup a Linux VM.

**Comment by eric-gitta-moore** [comment]
After a few hours of trying, I just lowered the docker desktop version to 4.21.1, after which he was fine. Then I upgraded to the latest version according to the prompts inside, and the failure still appeared.

TL;DR; Use version 4.21.2 of docker desktop
